### ISSUE TITLE
Very rare instance of not being able to remove bucket policy and same policy persisting even when the bucket was deleted and recreated

### ISSUE THREAD

**Issue by kaankabalak** [minio/minio#5717]
## Expected Behavior

The user should be able to remove bucket policies and the same policy should not appear when the user deletes the bucket that the policy has been set in.

## Current Behavior

On a very rare instance, for a small set of buckets where the bucket policy was set earlier, I couldn't remove bucket policy (I got an error while trying to remove it) and the same policy was still there even when the bucket was deleted and created again with the same name. However for the buckets that were created afterwards, this issue wasn't present.                      

## Possible Solution

This problem might have to do with caching, @krishnasrinivas told me that he will investigate.

## Steps to Reproduce (for bugs)

I couldn't reproduce the issue, but it happened when I was testing PR #5714. I left the server running while I switched branches and made bucket policy changes. Then I stopped the server, ran `go build` and started running the server again. I started to see this problem afterwards and it only stopped when I restarted the server.

## Context

Even though this was a one-off issue, it would cause problems for people who would like to change policies that were set in buckets.

## Your Environment

* Version used (`minio version`):
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`):
* Link to your project:

**Comment by harshavardhana** [comment]
isn't this issue same as https://github.com/minio/minio/issues/4498

**Comment by kaankabalak** [comment]
I don't think this is the same issue, as over there we were able to remove the existing bucket policy via the UI, but in this case we weren't able to remove the bucket policy at all.

**Comment by kaankabalak** [comment]
I don't think this is the same issue, as we changed that part of the code for the New UX on #5580. Also I was switching between the newux branch with the new design that was pushed to master and the branch where we reverted the changes, so it probably isn't the case that the old code caused the bug here. Additionally, on #4498, we were able to temporarily remove the existing bucket policies via the UI (such that there were no buckets on the policy list), but in this case we weren't able to remove them at all.
